wallet-new/src/Cardano/Wallet/WalletLayer/Types.hs

```diff
@@ -1,20 +1,53 @@
 module Cardano.Wallet.WalletLayer.Types
     ( PassiveWalletLayer (..)
     , ActiveWalletLayer (..)
+    -- * Lenses
+    , pwlCreateWallet
+    , pwlGetWalletIds
+    , pwlGetWallet
+    , pwlUpdateWallet
+    , pwlDeleteWallet
+
+    , pwlCreateAccount
+    , pwlGetAccounts
+    , pwlGetAccount
+    , pwlUpdateAccount
+    , pwlDeleteAccount
+
+    , pwlGetAddresses
     ) where
 
-import           Universum ()
+import           Universum
 
-import           Cardano.Wallet.API.V1.Types (WalletId)
+import           Control.Lens (makeLenses)
+
+import           Cardano.Wallet.API.V1.Types (Account, AccountIndex, AccountUpdate, Address,
+                                              NewAccount, NewWallet, Wallet, WalletId, WalletUpdate)
 
 import           Cardano.Wallet.Kernel.Diffusion (WalletDiffusion (..))
 
 
 -- | The passive wallet (data) layer. See @PassiveWallet@.
 data PassiveWalletLayer m = PassiveWalletLayer
-    { pwlGetWalletIds       :: m [WalletId]
+    {
+    -- * wallets
+      _pwlCreateWallet  :: NewWallet -> m Wallet
+    , _pwlGetWalletIds  :: m [WalletId]
+    , _pwlGetWallet     :: WalletId -> m (Maybe Wallet)
```

Comment:
As said by multiple folks, it would be better to have `Either LayerError a` rather than an opaque maybe.

Reply:
@ksaric Our data record should really be independent from any concrete monad instantiation -- how do you know that you will have `MonadThrow` for the _new_ data layer? I might be missing something, but that is not something which would be necessarily true, right? That depends from the particular `bracketXXX` implementation.

The less constraints on our monads, and the more explicit semantic in our pure data types, the better, imho 😉

Reply:
I agree.

> how do you know that you will have MonadThrow for the new data layer?

The same way I know we are going to use `Either` here - `Either` is forcing all the data layers to handle exceptions, and we could say the same for `MonadThrow`, I guess :man_shrugging: